### DISCUSSION TITLE
Randomly display activity posts on home page with buddypress

### DISCUSSION THREAD

**Discussion by wmasat** [community/community#62362] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

0

I have a buddypress, I want to display activity posts randomly on home page, not in chronological order. Anyone knows any code or method for that?

I used this code but it doesn't work buddypress
function posts_random_order( $query ) {

    if ( $query->is_main_query() && $query->is_home() ) {

        $query->set( 'orderby', 'rand' );

    }

}

add_action( 'pre_get_posts', 'posts_random_order' );

**Comment by nirav-gajera** [comment]
try this :

```
function posts_random_order( $query ) {
    if ( function_exists( 'bp_is_activity_front_page' ) && bp_is_activity_front_page() && $query->is_main_query() ) {
        $query->set( 'orderby', 'rand' );
    }
}
add_action( 'pre_get_posts', 'activity_random_order' );

```

This code checks if the current page is the BuddyPress activity front page using the `bp_is_activity_front_page()` function and then sets the orderby parameter to 'rand' to display the activity posts randomly.

**Comment by wmasat** [comment]
> try this :
> 
> ```
> function posts_random_order( $query ) {
>     if ( function_exists( 'bp_is_activity_front_page' ) && bp_is_activity_front_page() && $query->is_main_query() ) {
>         $query->set( 'orderby', 'rand' );
>     }
> }
> add_action( 'pre_get_posts', 'activity_random_order' );
> ```
> 
> This code checks if the current page is the BuddyPress activity front page using the `bp_is_activity_front_page()` function and then sets the orderby parameter to 'rand' to display the activity posts randomly.

**Reply by wmasat** [reply]
Unfortunately, it does not work. Can you modify the code?
